lib/query_relation/array_based_collection.rb

```diff
@@ -0,0 +1,14 @@
+class QueryRelation
+  module ArrayBasedCollection
+    # Default filter for an array based +collection+ based on various hash
+    # based +options+ that are used by QueryRelation.
+    #
+    def filter_array_collection(collection, options)
+      collection = collection.drop(options[:offset]) if options[:offset]
+      collection = collection.take(options[:limit]) if options[:limit]
```

Comment:
NICE!

I always use something like `collection[(options[:offset] || 0)..(options[:limit] || -1)]`
`take` and `drop` seems much more efficient

Reply:
@Fryguy gets the credit for that, I just copy/pasted that. :)